lib/mkmf.rb

```diff
@@ -2148,6 +2148,7 @@ def configuration(srcdir)
 DLDFLAGS = $(ldflags) $(dldflags) $(ARCH_FLAG)
 LDSHARED = #{CONFIG['LDSHARED']}
 LDSHAREDXX = #{config_string('LDSHAREDXX') || '$(LDSHARED)'}
+POSTLINK = #{RbConfig.expand(CONFIG['POSTLINK'], CONFIG)}
```

Comment:
I used `RbConfig.expand` here to avoid carrying all possible referenced variables like `WASMOPT`, `wasmoptflags`, and so on.